### ISSUE TITLE
[Compatibility] Writing to missing Chunk

### ISSUE THREAD

**Issue by SharkieTV** [paulevsGitch/BetterNether#477]
### What happened?

Crashing on Better Minecraft Fabric Modpack

### BetterNether

better-nether-6.0.7.jar

### BCLib

bclib-1.1.3.jar

### Fabric API

fabric-api-0.44.0+1.18.jar

### Fabric Loader

0.12.12

### Minecraft

1.18.1

### Relevant log output

```shell
Crash - https://pastebin.com/nRKgBiiw
```

### Other Mods

```shell
You can also see all the Mods in here - https://pastebin.com/nRKgBiiw
```

**Comment by quiqueck** [comment]
Couly you please provide more details. What were you doing at the time of the crash? Do you maybe happen to have your world pos at the time of the crash (that may be related to the issue). Is the crash reproducible as in, it happens every time you load your level?

**Comment by SharkieTV** [comment]
This is from my players from my modpack and I cant recreate this but I think its when they first loaded the nether and It seems to be a feature placement issue. Amplified Nether is installed but that shouldnt cause the issue?

**Comment by quiqueck** [comment]
Thanks for the quick reply. I am not sure yet. I think Amplified Nether alone does not cause this issue as I am testing most new releases with that one. I'll have to install Better Fabric in my testing environment. Hope I can reproduce the error with it :)

**Comment by SharkieTV** [comment]
Here are another 2 Nether mods that does some world generation in the nether
Emerald Geodes (+now with Quartz!)
Lava Java

**Comment by SharkieTV** [comment]
actually now that I have a better look it seems like he added mods that is not in the pack, I think there is no issue they created it by adding mods

**Comment by quiqueck** [comment]
Definetly. We had a similar report a while back, so there might be a thing. Do you know which mods they added?

**Comment by SharkieTV** [comment]
holograms: Holograms 1.3.1+1.18
ec-core: Essential Commands Core 1.0.1-mc1.18.1
elativitymc_c2me-fabric_ver_1_18_libs_: libs 0.2.0+alpha.5.106
de_erdbeerbaerlp_discordintegration-core: DiscordIntegration-Core ce44c82e
dcintegration-fabric: Discord Integration Fabric 2.3.6
confabricate: confabricate 2.2.0-SNAPSHOT+4.1.1
com_velocitypowered_velocity-native: velocity-native 1.1.0-SNAPSHOT
c2me: Concurrent Chunk Management Engine 0.2.0+alpha.5.106
blue_endless_jankson: jankson 1.2.1
packet_tweaker: Packet Tweaker 0.2.0+1.18-pre1
minimotd-fabric: MiniMOTD 2.0.5
mm: Manningham Mills 2.3

Now I only took these from the log since the person isn't responding but, idk if these aren't even mods but they don't seem familiar and there might be more mods too

**Comment by quiqueck** [comment]
That looks interesting `Concurrent Chunk Management Engine`

**Comment by quiqueck** [comment]
Yep, I can confirm: `Concurrent Chunk Management Engine` is almost instantly triggering that error for me :)

**Comment by quiqueck** [comment]
Ah, and they already have an issue for it too: RelativityMC/C2ME-fabric#97

**Comment by SharkieTV** [comment]
kk thanks for that, so there is no issue with the better nether mod right?

**Comment by quiqueck** [comment]
NO, that mod changes a core behaviour of vanilla Minecraft which introduces sideeffects like those.

In more Detail: Well, we are not prepared for that kind of paralell Chunk processing the mod forces on the game (to increase the CPU usage which will speed up load-times, which is great). The last commit already fixed the issue you reported. But now I get new ones in very basic and frequently called methods. If I fix those, it will have a huge speed impact for all users (beeing safe for parallel execution introduces cost), so I am not sure if we can make it work. But I am still investigating.

Either way, its is definetly not going to be a quick fix. Oh, and btw. thanks for a great ModPack ;)

**Comment by SharkieTV** [comment]
Awesome thanks, cant wait for Better End to come out!

**Comment by quiqueck** [comment]
This should be fixed in 6.0.8. Please  reopen or start a new issue if the problem persists.
